cpp/src/arrow/dataset/file_base.cc

```diff
@@ -519,7 +519,7 @@ class TeeNode : public compute::MapNode {
       MapNode::Finish(std::move(writer_finish_st));
       return;
     }
-    serial_scheduler_->End();
```

Comment:
It almost feels like sub-schedulers should be a different interface (or parent interface) of the 'main' scheduler so that these methods don't even exist for clients (though this is not a big deal).

Reply:
I suppose since we plan to get rid of End() entirely, then this is moot.